janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -6165,4 +6165,251 @@ private void testWritingAndReading(File f) {
+    	assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+    	assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+    	assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+    	assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+    	
+    	getV(graph, a).remove();
+        getV(graph, b).remove();
+    	
+    	close();
+    }
+    
+    
+    @Test
+    public void testReindexingForEdgeIndexDirectionBOTH() throws InterruptedException, ExecutionException {
+    	clopen(option(LOG_SEND_DELAY, MANAGEMENT_LOG), Duration.ofMillis(0),
+                option(KCVSLog.LOG_READ_LAG_TIME, MANAGEMENT_LOG), Duration.ofMillis(50),
+                option(LOG_READ_INTERVAL, MANAGEMENT_LOG), Duration.ofMillis(250)
+        );
+    	
+    	String indexName = "edgesByAssocKindBoth";
+    	
+    	//Schema creation
+    	initReindexSchema();
+    	EdgeLabel edgeLabel = mgmt.getEdgeLabel("egLabel");
+    	PropertyKey propAssocKind = mgmt.getPropertyKey("assocKind");
+    	mgmt.buildEdgeIndex(edgeLabel, indexName, Direction.BOTH, propAssocKind);
+    	finishSchema();
+    	
+    	//Create Vertex
+    	JanusGraphVertex a = tx.addVertex();
+    	a.property("vtName","A");
+    	JanusGraphVertex b = tx.addVertex();
+    	b.property("vtName","B");
+    	JanusGraphVertex c = tx.addVertex();
+    	c.property("vtName","C");
+    	
+    	//Add Edges
+    	a.addEdge("egLabel",b,"assocKind",1);
+    	a.addEdge("egLabel",c,"assocKind",1);
+    	c.addEdge("egLabel",a,"assocKind",1);
+    	c.addEdge("egLabel",b,"assocKind",1);
+    	newTx();
+
+    	RelationType t = mgmt.getRelationType("egLabel");
+    	RelationTypeIndex relationIndex = mgmt.getRelationIndex(t,indexName);
+    	assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+    	
+    	//asserting before reindex
+    	assertEdgeCounts(false);
+    	
+    	//Reindexing
+    	mgmt.updateIndex(relationIndex, SchemaAction.REINDEX).get();
+    	finishSchema();
+    	
+    	relationIndex = mgmt.getRelationIndex(t,indexName);
+    	assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+    	
+    	//asserting after reindex
+    	assertEdgeCounts(true);
+    	
+    	GraphTraversalSource g = graph.traversal();
+    	Vertex aV = g.V().has("vtName" , "A").next();
+    	Vertex bV = g.V().has("vtName" , "B").next();
+    	Vertex cV = g.V().has("vtName" , "C").next();
+    	//assert the vertices of the given relation
+    	GraphTraversal<Vertex, Edge> has = g.V().has("vtName" , "C").inE().hasLabel("egLabel").has("assocKind",1);
+    	Edge edge = has.next();
+    	assertEquals(1,edge.property("assocKind").value());
+    	assertEquals("egLabel",edge.label());
+    	assertEquals(cV.id(), edge.inVertex().id());
+    	assertEquals(aV.id(), edge.outVertex().id());
+    	
+    	getV(graph, a).remove();
```

Comment:
Why do you need to run remove?

Reply:
Im creating the vertex with same properties in other testcases also to test index with different directions. Hence removing the vertex after completion of every testcase to avoid conflicts.

Reply:
Tests in this class should be independent.